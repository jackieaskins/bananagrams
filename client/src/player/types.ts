export type Player = {
  userId: string;
  username: string;
  isReady: boolean;
  isTopBanana: boolean;
  isAdmin: boolean;
  gamesWon: number;
};
