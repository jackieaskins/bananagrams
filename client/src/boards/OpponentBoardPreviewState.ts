import { useEffect, useState } from 'react';
import { Player } from '../players/types';

type OpponentBoardPreviewState = {
  handleLeftClick: () => void;
  handleRightClick: () => void;
  handleSelectedPlayerChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  selectedPlayerIndex: number;
  selectedUserId?: string;
};

export const useOpponentBoardPreview = (
  opponents: Player[],
  initialPlayerIndex: number
): OpponentBoardPreviewState => {
  const [selectedUserId, setSelectedUserId] = useState<string | undefined>(
    opponents[initialPlayerIndex]?.userId
  );

  const foundIndex = opponents.findIndex(
    ({ userId }) => userId === selectedUserId
  );
  const selectedPlayerIndex = foundIndex < 0 ? 0 : foundIndex;

  useEffect(() => {
    if (!opponents.some(({ userId }) => userId === selectedUserId)) {
      setSelectedUserId(opponents[0]?.userId);
    }
  }, [opponents, selectedUserId]);

  const handleLeftClick = (): void => {
    const index =
      selectedPlayerIndex <= 0 ? opponents.length - 1 : selectedPlayerIndex - 1;
    setSelectedUserId(opponents[index]?.userId);
  };

  const handleRightClick = (): void => {
    const index =
      selectedPlayerIndex >= opponents.length - 1 ? 0 : selectedPlayerIndex + 1;
    setSelectedUserId(opponents[index]?.userId);
  };

  const handleSelectedPlayerChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    setSelectedUserId(e.target.value);
  };

  return {
    handleLeftClick,
    handleRightClick,
    handleSelectedPlayerChange,
    selectedPlayerIndex,
    selectedUserId,
  };
};
