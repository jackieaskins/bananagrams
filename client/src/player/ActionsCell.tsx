import { DeleteOutlined } from '@ant-design/icons';
import { Button } from 'antd';

import { useCurrentPlayer } from '../game/stateHooks';
import { kickPlayer } from '../socket';
import { Player } from './types';

type ActionsCellProps = {
  player: Player;
};

const ActionsCell = ({
  player: { userId },
}: ActionsCellProps): JSX.Element | null => {
  const currentPlayer = useCurrentPlayer();

  if (!currentPlayer) {
    return null;
  }

  const { isAdmin, userId: currentPlayerId } = currentPlayer;

  if (isAdmin && userId !== currentPlayerId) {
    return (
      <Button
        icon={<DeleteOutlined />}
        onClick={() => {
          kickPlayer({ userId });
        }}
        type="text"
      />
    );
  }

  return null;
};

export default ActionsCell;
