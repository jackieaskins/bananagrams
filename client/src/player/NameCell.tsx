import { CrownOutlined, KeyOutlined } from '@ant-design/icons';
import { Space } from 'antd';

import { Player } from '../players/types';

type NameCellProps = {
  player: Player;
};

const NameCell = ({
  player: { isAdmin, isTopBanana, username },
}: NameCellProps): JSX.Element => (
  <Space>
    <span>{username}</span>
    {isAdmin && (
      <span>
        <KeyOutlined />
      </span>
    )}
    {isTopBanana && (
      <span>
        <CrownOutlined />
      </span>
    )}
  </Space>
);

export default NameCell;
