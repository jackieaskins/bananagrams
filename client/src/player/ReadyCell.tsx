import { CheckOutlined, CloseOutlined } from '@ant-design/icons';
import { Checkbox } from 'antd';

import { getUserId, setReady } from '../socket';
import { Player } from './types';

type ReadyCellProps = {
  player: Player;
};

const ReadyCell = ({
  player: { userId, isReady },
}: ReadyCellProps): JSX.Element => {
  const isCurrentUser = userId === getUserId();

  if (isCurrentUser) {
    return (
      <Checkbox
        checked={isReady}
        onChange={({ target: { checked } }) => {
          setReady({ isReady: checked });
        }}
      />
    );
  }

  if (isReady) {
    return <CheckOutlined css={{ color: 'green' }} />;
  }

  return <CloseOutlined css={{ color: 'red' }} />;
};
export default ReadyCell;
