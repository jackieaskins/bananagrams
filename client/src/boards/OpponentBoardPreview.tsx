import PreviewBoard from './PreviewBoard';
import { useSocket } from '../socket/SocketContext';
import { Box, Grid, IconButton, MenuItem, TextField } from '@material-ui/core';
import { NavigateBefore, NavigateNext } from '@material-ui/icons';
import PreviewHand from '../hands/PreviewHand';
import { Player } from '../players/types';
import { useOpponentBoardPreview } from './OpponentBoardPreviewState';

type OpponentBoardPreviewProps = {
  initialPlayerIndex?: number;
  players: Player[];
  tileSize?: number;
  includeCurrentPlayer?: boolean;
};

const EMPTY_BOARD = [...Array(21)].map(() => Array(21).fill(null));

const OpponentBoardPreview: React.FC<OpponentBoardPreviewProps> = ({
  initialPlayerIndex = 0,
  players,
  tileSize = 15,
  includeCurrentPlayer = false,
}) => {
  const {
    socket: { id: userId },
  } = useSocket();

  const opponents = includeCurrentPlayer
    ? players
    : players.filter((player) => player.userId !== userId);
  const {
    handleLeftClick,
    handleRightClick,
    handleSelectedPlayerChange,
    selectedPlayerIndex,
    selectedUserId,
  } = useOpponentBoardPreview(opponents, initialPlayerIndex);

  if (opponents.length === 0) {
    return null;
  }

  const selectedOpponent = opponents[selectedPlayerIndex];
  const selectedBoard = selectedOpponent?.board ?? EMPTY_BOARD;
  const selectedHand = selectedOpponent?.hand ?? [];
  const hasOneOpponent = opponents.length === 1;

  return (
    <Grid container direction="column" alignItems="center" spacing={1}>
      <Grid item>
        <Box flexDirection="column" width="317px">
          <PreviewBoard board={selectedBoard} tileSize={tileSize} />
          <PreviewHand hand={selectedHand} tileSize={tileSize} />
        </Box>
      </Grid>

      <Grid item style={{ width: '100%' }}>
        <Box display="flex" justifyContent="space-between">
          <IconButton
            size="small"
            disabled={hasOneOpponent}
            onClick={handleLeftClick}
          >
            <NavigateBefore fontSize="small" />
          </IconButton>

          <TextField
            select
            size="small"
            value={selectedUserId}
            onChange={handleSelectedPlayerChange}
            disabled={hasOneOpponent}
          >
            {opponents.map(({ userId, username }) => (
              <MenuItem value={userId} key={userId}>
                {username}
              </MenuItem>
            ))}
          </TextField>

          <IconButton
            size="small"
            disabled={hasOneOpponent}
            onClick={handleRightClick}
          >
            <NavigateNext fontSize="small" />
          </IconButton>
        </Box>
      </Grid>
    </Grid>
  );
};

export default OpponentBoardPreview;
