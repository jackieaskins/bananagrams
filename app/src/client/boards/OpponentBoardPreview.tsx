import { HStack, IconButton, Select, Stack, Text } from "@chakra-ui/react";
import { useState, useEffect, useMemo, useCallback } from "react";
import { FaChevronLeft, FaChevronRight } from "react-icons/fa6";
import PlayerPreview from "@/client/redesign/PlayerPreview";
import { socket } from "@/client/socket";
import { Board } from "@/types/board";
import { Player } from "@/types/player";

type OpponentBoardPreviewProps = {
  initialPlayerIndex?: number;
  players: Player[];
  tileSize?: number;
  includeCurrentPlayer?: boolean;
};

const EMPTY_BOARD: Board = {};

export default function OpponentBoardPreview({
  initialPlayerIndex = 0,
  players,
  tileSize = 15,
  includeCurrentPlayer = false,
}: OpponentBoardPreviewProps): JSX.Element | null {
  const opponents = includeCurrentPlayer
    ? players
    : players.filter((player) => player.userId !== socket.id);
  const [selectedUserId, setSelectedUserId] = useState<string | undefined>(
    opponents[initialPlayerIndex]?.userId,
  );

  useEffect(() => {
    if (!opponents.some(({ userId }) => userId === selectedUserId)) {
      setSelectedUserId(opponents[0]?.userId);
    }
  }, [opponents, selectedUserId]);

  const selectedPlayerIndex = useMemo(() => {
    const foundIndex = opponents.findIndex(
      ({ userId }) => userId === selectedUserId,
    );
    return foundIndex < 0 ? 0 : foundIndex;
  }, [opponents, selectedUserId]);

  const handleLeftClick = useCallback(() => {
    const index =
      selectedPlayerIndex <= 0 ? opponents.length - 1 : selectedPlayerIndex - 1;
    setSelectedUserId(opponents[index]?.userId);
  }, [opponents, selectedPlayerIndex]);

  const handleRightClick = useCallback(() => {
    const index =
      selectedPlayerIndex >= opponents.length - 1 ? 0 : selectedPlayerIndex + 1;
    setSelectedUserId(opponents[index]?.userId);
  }, [opponents, selectedPlayerIndex]);

  const handleSelectedPlayerChange = useCallback(
    (e: React.ChangeEvent<HTMLSelectElement>) => {
      setSelectedUserId(e.target.value);
    },
    [],
  );

  if (opponents.length === 0) {
    return null;
  }

  const selectedOpponent = opponents[selectedPlayerIndex];
  const selectedBoard = selectedOpponent?.board ?? EMPTY_BOARD;
  const selectedHand = selectedOpponent?.hand ?? [];
  const hasOneOpponent = opponents.length === 1;

  return (
    <Stack width="300px">
      <PlayerPreview
        board={selectedBoard}
        hand={selectedHand}
        tileSize={tileSize}
      />

      <HStack spacing={4} width="100%">
        {hasOneOpponent ? (
          <Text textAlign="center" width="100%">
            {opponents[0].username}
          </Text>
        ) : (
          <>
            <IconButton
              aria-label="See previous opponent's board"
              icon={<FaChevronLeft />}
              onClick={handleLeftClick}
              variant="outline"
              size="sm"
            />

            <Select
              value={selectedUserId}
              onChange={handleSelectedPlayerChange}
              size="sm"
            >
              {opponents.map(({ userId, username }) => (
                <option value={userId} key={userId}>
                  {username}
                </option>
              ))}
            </Select>

            <IconButton
              aria-label="See next opponent's board"
              icon={<FaChevronRight />}
              onClick={handleRightClick}
              variant="outline"
              size="sm"
            />
          </>
        )}
      </HStack>
    </Stack>
  );
}
