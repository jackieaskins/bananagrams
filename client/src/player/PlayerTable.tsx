import { Table, Typography } from 'antd';
import { ColumnType } from 'antd/lib/table';

import { useCurrentPlayer, useGamePlayers } from '../game/stateHooks';
import ActionsCell from './ActionsCell';
import NameCell from './NameCell';
import ReadyCell from './ReadyCell';
import { Player } from './types';

const COLUMN_ALIGN = 'center';
const MAX_PLAYERS = 8;

const { Text } = Typography;

const actionsColumn: ColumnType<Player> = {
  align: COLUMN_ALIGN,
  key: 'actions',
  render: function ActionsRender(_: any, player: Player) {
    return <ActionsCell player={player} />;
  },
};

const PlayerTable = (): JSX.Element => {
  const players = useGamePlayers();
  const isAdmin = !!useCurrentPlayer()?.isAdmin;

  return (
    <Table
      bordered
      columns={[
        {
          title: 'Players',
          children: [
            {
              align: COLUMN_ALIGN,
              key: 'ready',
              title: 'Ready?',
              render: function ReadyRender(_, player) {
                return <ReadyCell player={player} />;
              },
            },
            {
              align: COLUMN_ALIGN,
              key: 'player',
              title: 'Player',
              render: function PlayerRender(_, player) {
                return <NameCell player={player} />;
              },
            },
            {
              align: COLUMN_ALIGN,
              dataIndex: 'gamesWon',
              title: 'Games won',
            },
            ...(isAdmin && players.length > 1 ? [actionsColumn] : []),
          ],
        },
      ]}
      dataSource={players}
      footer={() => (
        <div css={{ fontSize: '12px', textAlign: 'center' }}>
          <Text type="secondary">
            The game will start when all players are ready
            <br />
            Current player count: {players.length} / {MAX_PLAYERS}
          </Text>
        </div>
      )}
      pagination={false}
      rowKey="userId"
      size="middle"
    />
  );
};

export default PlayerTable;
