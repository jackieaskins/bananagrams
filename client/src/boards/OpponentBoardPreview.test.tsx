import { shallow } from 'enzyme';

import { playerFixture } from '../fixtures/player';
import OpponentBoardPreview from './OpponentBoardPreview';
import { useOpponentBoardPreview } from './OpponentBoardPreviewState';

jest.mock('../socket/SocketContext', () => ({
  useSocket: () => ({
    socket: { id: 'playerId' },
  }),
}));

jest.mock('./OpponentBoardPreviewState', () => ({
  useOpponentBoardPreview: jest.fn(),
}));

describe('<OpponentBoardPreview />', () => {
  const renderComponent = (propOverrides = {}) =>
    shallow(
      <OpponentBoardPreview
        players={[playerFixture({ userId: 'playerId' })]}
        hands={{ opponent: [], playerId: [] }}
        boards={{ opponent: [[null]], playerId: [[null]] }}
        {...propOverrides}
      />
    );

  const mockOpponentBoardPreviewState = (
    selectedPlayerIndex = 0,
    selectedUserId = 'opponent'
  ) => {
    useOpponentBoardPreview.mockReturnValue({
      handleLeftClick: jest.fn().mockName('handleLeftClick'),
      handleRightClick: jest.fn().mockName('handleRightClick'),
      handleSelectedPlayerChange: jest
        .fn()
        .mockName('handleSelectedPlayerChange'),
      selectedPlayerIndex,
      selectedUserId,
    });
  };

  beforeEach(() => {
    mockOpponentBoardPreviewState();
  });

  test('does not render if no opponents', () => {
    expect(renderComponent()).toMatchSnapshot();
  });

  test('renders current player if includeCurrentPlayer', () => {
    expect(
      renderComponent({
        includeCurrentPlayer: true,
        players: [playerFixture({ userId: 'playerId' })],
      })
    ).toMatchSnapshot();
  });

  test('renders properly with one opponent', () => {
    expect(
      renderComponent({ players: [playerFixture({ userId: 'opponent' })] })
    ).toMatchSnapshot();
  });

  test('renders properly with multiple opponents', () => {
    expect(
      renderComponent({
        players: [
          playerFixture({ userId: 'opponent' }),
          playerFixture({ userId: 'opponent2' }),
        ],
      })
    ).toMatchSnapshot();
  });

  test('renders empty board and hand when selectedIndex is out of bounds', () => {
    mockOpponentBoardPreviewState(5);

    expect(
      renderComponent({
        players: [playerFixture({ userId: 'opponent' })],
      })
    ).toMatchSnapshot();
  });
});
