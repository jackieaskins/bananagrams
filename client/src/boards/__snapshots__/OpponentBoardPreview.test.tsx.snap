// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpponentBoardPreview /> does not render if no opponents 1`] = `""`;

exports[`<OpponentBoardPreview /> renders current player if includeCurrentPlayer 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      flexDirection="column"
      width="317px"
    >
      <PreviewBoard
        board={
          Array [
            Array [
              null,
            ],
          ]
        }
        tileSize={15}
      />
      <PreviewHand
        hand={Array []}
        tileSize={15}
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Styled(MuiBox)
      display="flex"
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(IconButton))
        disabled={true}
        onClick={[MockFunction handleLeftClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateBeforeIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(TextField))
        disabled={true}
        onChange={[MockFunction handleSelectedPlayerChange]}
        select={true}
        size="small"
        value="opponent"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="playerId"
          value="playerId"
        >
          username
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(TextField))>
      <WithStyles(ForwardRef(IconButton))
        disabled={true}
        onClick={[MockFunction handleRightClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateNextIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`<OpponentBoardPreview /> renders empty board and hand when selectedIndex is out of bounds 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      flexDirection="column"
      width="317px"
    >
      <PreviewBoard
        board={
          Array [
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            Array [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
          ]
        }
        tileSize={15}
      />
      <PreviewHand
        hand={Array []}
        tileSize={15}
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Styled(MuiBox)
      display="flex"
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(IconButton))
        disabled={true}
        onClick={[MockFunction handleLeftClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateBeforeIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(TextField))
        disabled={true}
        onChange={[MockFunction handleSelectedPlayerChange]}
        select={true}
        size="small"
        value="opponent"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="opponent"
          value="opponent"
        >
          username
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(TextField))>
      <WithStyles(ForwardRef(IconButton))
        disabled={true}
        onClick={[MockFunction handleRightClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateNextIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`<OpponentBoardPreview /> renders properly with multiple opponents 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      flexDirection="column"
      width="317px"
    >
      <PreviewBoard
        board={
          Array [
            Array [
              null,
            ],
          ]
        }
        tileSize={15}
      />
      <PreviewHand
        hand={Array []}
        tileSize={15}
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Styled(MuiBox)
      display="flex"
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(IconButton))
        disabled={false}
        onClick={[MockFunction handleLeftClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateBeforeIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(TextField))
        disabled={false}
        onChange={[MockFunction handleSelectedPlayerChange]}
        select={true}
        size="small"
        value="opponent"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="opponent"
          value="opponent"
        >
          username
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          key="opponent2"
          value="opponent2"
        >
          username
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(TextField))>
      <WithStyles(ForwardRef(IconButton))
        disabled={false}
        onClick={[MockFunction handleRightClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateNextIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`<OpponentBoardPreview /> renders properly with one opponent 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      flexDirection="column"
      width="317px"
    >
      <PreviewBoard
        board={
          Array [
            Array [
              null,
            ],
          ]
        }
        tileSize={15}
      />
      <PreviewHand
        hand={Array []}
        tileSize={15}
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Styled(MuiBox)
      display="flex"
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(IconButton))
        disabled={true}
        onClick={[MockFunction handleLeftClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateBeforeIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(TextField))
        disabled={true}
        onChange={[MockFunction handleSelectedPlayerChange]}
        select={true}
        size="small"
        value="opponent"
      >
        <WithStyles(ForwardRef(MenuItem))
          key="opponent"
          value="opponent"
        >
          username
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(TextField))>
      <WithStyles(ForwardRef(IconButton))
        disabled={true}
        onClick={[MockFunction handleRightClick]}
        size="small"
      >
        <Memo(ForwardRef(NavigateNextIcon))
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
